### PR TITLE
Ensure Dynamo builds on Linux, Mac

### PR DESCRIPTION
**Pull request by pboyer**
### Issue

Dynamo's csproj files are riddled with platform specific code.  This PR fixes that.  After this fix, you should be able to build Dynamo on *nix plaforms with a single command:

    # xbuild Dynamo.Mono.sln

You can see the successful build on Travis CI here:

https://travis-ci.org/pboyer/Dynamo​

These are the code changes required to complete:

https://github.com/DynamoDS/Dynamo/wiki/WIP:-Cross-platform-development-with-Dynamo

### Fix

- [x] Add new Dynamo.Mono.sln file that excludes Windows projects
- [x] Remove legacy *Mono.csproj projects
- [x] Where possible, check for the platform or use a cross-platform copy Task in csproj files
- [x] Confirm builds correctly on *Nix platforms
- [x] Confirm builds correctly on Windows
- [x] Confirm tests run successfully on Windows, Linux
- [x] Update wiki documentation

### Reviewer

- [ ] @lukechurch 

FYI @brandoncole

### CODE REVIEW

File: src/AssemblySharedInfoGenerator/AssemblyInfoGenerator.csproj [hunk 1]
```diff
@@ -63,7 +63,7 @@
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
   <PropertyGroup>
-    <PreBuildEvent>"$(SolutionDir)transform_all.bat" "$(ProjectDir)"</PreBuildEvent>
+    <PreBuildEvent Condition=" '$(OS)' != 'Unix' ">"$(SolutionDir)transform_all.bat" "$(ProjectDir)"</PreBuildEvent>
   </PropertyGroup>
```
Comment (pboyer) [inline src/AssemblySharedInfoGenerator/AssemblyInfoGenerator.csproj]: This operation is not done on *Nix platforms as it requires TT template transforms to be installed.

File: src/Legacy/ProtoTestExe/ProtoTestExeMono.csproj [hunk 2]
```diff
@@ -1,126 +0,0 @@
...
-    </ProjectReference>
-  </ItemGroup>
-  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-  <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
-       Other similar extension points exist, see Microsoft.Common.targets.
-  <Target Name="BeforeBuild">
-  </Target>
-  <Target Name="AfterBuild">
-  </Target>
-  -->
```
Comment (pboyer) [inline src/Legacy/ProtoTestExe/ProtoTestExeMono.csproj]: I deleted these *Mono.csproj files as they are not in use.  Are they necessary?

File: src/Libraries/DSIronPython/IronPythonEval.csproj [hunk 3]
```diff
@@ -1,79 +1,79 @@
...
+	  <ItemGroup>
+	    <Content Include="DSIronPython_DynamoCustomization.xml" />
+	  </ItemGroup>
+	  <ItemGroup>
+	    <ProjectReference Include="..\..\DynamoUtilities\DynamoUtilities.csproj">
+	      <Project>{b5f435cb-0d8a-40b1-a4f7-5ecb3ce792a9}</Project>
+	      <Name>DynamoUtilities</Name>
+	    </ProjectReference>
+	  </ItemGroup>
+	  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
```
Comment (pboyer) [inline src/Libraries/DSIronPython/IronPythonEval.csproj]: I don't know how I managed to introduce these new lines.  The only actual changes are below.

File: src/Engine/ProtoCore/ProtoCore.csproj [hunk 4]
```diff
@@ -112,7 +112,7 @@
     <Compile Include="DebuggerProperties.cs" />
     <Compile Include="Diagnostics\RuntimeDiagnostics.cs" />
     <Compile Include="DSASM\FunctionCounter.cs" />
-    <Compile Include="DSASM\DSAsmDefs.cs" />
+    <Compile Include="DSASM\DSasmDefs.cs" />
```
Comment (pboyer) [inline src/Engine/ProtoCore/ProtoCore.csproj]: Unix is case-sensitive.

File: src/Dynamo.Mono.sln [hunk 5]
```diff
@@ -0,0 +1,433 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio 2013
```
Comment (lukechurch) [inline src/Dynamo.Mono.sln]: Does this need to be named Mono.2013?

File: src/DynamoCore/DynamoCore.csproj [hunk 6]
```diff
@@ -329,33 +329,29 @@ limitations under the License.
   </ItemGroup>
   <ItemGroup />
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-  <PropertyGroup>
-    <PostBuildEvent>copy /Y "$(SolutionDir)..\extern\DesignScript\*" "$(OutputPath)"
-copy "$(SolutionDir)..\doc\distrib\License.rtf" "$(OutputPath)\$(UICulture)\License.rtf"
-copy "$(SolutionDir)..\tools\install\extra\InstallASMForDynamo.exe" "$(OutputPath)\InstallASMForDynamo.exe"
-IF NOT EXIST "$(OutputPath)\dll" "$(OutputPath)\InstallASMForDynamo.exe" /SILENT /SUPPRESSMSGBOXES</PostBuildEvent>
```
Comment (lukechurch) [inline src/DynamoCore/DynamoCore.csproj]: Is there an effect of eliminating this call?

File: src/Libraries/CoreNodeModels/CoreNodeModels.csproj [hunk 7]
```diff
@@ -54,7 +54,7 @@
     <Reference Include="System.Xml" />
   </ItemGroup>
   <ItemGroup>
-    <Compile Include="DynamoConvert.cs" />
+    <Compile Condition=" '$(OS)' != 'Unix' " Include="DynamoConvert.cs" />
```
Comment (lukechurch) [inline src/Libraries/CoreNodeModels/CoreNodeModels.csproj]: Really? Why would DynamoConvert be platform specific?

File: test/runtests.sh [hunk 8]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh -x
+
+mono --runtime=v4.0 .nuget/NuGet.exe install NUnit.Runners -Version 2.6.1 -o packages
+
+runTest(){
+    mono --runtime=v4.0 packages/NUnit.Runners.2.6.1/tools/nunit-console.exe -noxml -nodots -labels -stoponerror $@
+   if [ $? -ne 0 ]
```
Comment (lukechurch) [inline test/runtests.sh]: This could use a comment.

**Comment by lukechurch** [comment]
lgtm with a couple of queries.

File: src/DynamoCore/DynamoCore.csproj (hunk 6 above)
Comment (pboyer) [inline src/DynamoCore/DynamoCore.csproj]: Actually those files are gone (months ago now), so I can remove it.  :)

File: src/Libraries/CoreNodeModels/CoreNodeModels.csproj (hunk 7 above)
Comment (pboyer) [inline src/Libraries/CoreNodeModels/CoreNodeModels.csproj]: I'll look into this.

**Pull request merged**